staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -116,8 +119,54 @@ service RuntimeService {
     rpc Status(StatusRequest) returns (StatusResponse) {}
 }
 
+// GetEventsFilter is used to filter a list of events.
+// All those fields are combined with 'AND'
+message GetEventsFilter {
+    // ID of the container, sandbox or image.
+    string id = 1;
+    // LabelSelector to select matches.
+    // Only api.MatchLabels is supported for now and the requirements
+    // are ANDed. MatchExpressions is not supported yet.
+    map<string, string> label_selector = 3;
+}
+
+message GetEventsRequest {
+    // Optional to filter a list of events.
+    GetEventsFilter filter = 1;
+}
+
+message ContainerEventResponse {
+    // ID of the container
+	string container_id = 1;
+
+	// Type of the container event
+    ContainerEventType container_event_type = 2;
+
+    // Creation timestamp of this event
+    int64 created_at = 3;
+
+    // ID of the sandbox container
+	string sandbox_id = 4;
+}
+
+enum ContainerEventType {
+    // Container created
+    CONTAINER_CREATED_EVENT = 0;
+
+    // Container started
+    CONTAINER_STARTED_EVENT = 1;
+
+    // Container stopped
+    CONTAINER_STOPPED_EVENT = 2;
+
+    // Container deleted
+    CONTAINER_DELETED_EVENT = 3;
+}
```

Comment:
Would it be premature to add a type CONTAINER_UPDATED (for resource update actuation notification and such - TBD in the future; out-of-scope for now)?

Reply:
We are aiming to reduce the current frequent relisting done by kubelet PLEG. If this helps in that direction, we could include it. If not, we could revisit this in the future.